Some of test_wrappers.py fail without -Werror

```
$ pytest -Wdefault
[...]
FAILED tests/test_wrappers.py::test_response_mixins_deprecated[BaseResponse] - Failed: DID NOT RAISE <class 'DeprecationWarning'>
FAILED tests/test_wrappers.py::test_response_mixins_deprecated[CommonResponseDescriptorsMixin] - Failed: DID NOT RAISE <class 'DeprecationWarni...
FAILED tests/test_wrappers.py::test_response_mixins_deprecated[ResponseStreamMixin] - Failed: DID NOT RAISE <class 'DeprecationWarning'>
FAILED tests/test_wrappers.py::test_response_mixins_deprecated[ETagResponseMixin] - Failed: DID NOT RAISE <class 'DeprecationWarning'>
FAILED tests/test_wrappers.py::test_response_mixins_deprecated[WWWAuthenticateMixin] - Failed: DID NOT RAISE <class 'DeprecationWarning'>
FAILED tests/test_wrappers.py::test_response_mixins_deprecated[CORSResponseMixin] - Failed: DID NOT RAISE <class 'DeprecationWarning'>
FAILED tests/test_wrappers.py::test_response_mixins_deprecated[JSONMixin] - Failed: DID NOT RAISE <class 'DeprecationWarning'>
FAILED tests/test_wrappers.py::test_check_base_deprecated - Failed: DID NOT RAISE <class 'DeprecationWarning'>
FAILED tests/test_wrappers.py::test_response_freeze_no_etag_deprecated - Failed: DID NOT RAISE <class 'DeprecationWarning'>
```

While I realize that my approach may seem unsupported, we find it really useful to test whether packages work on end user systems. In this case, it is only important that the package *works*, deprecation warnings are fine for us since we aren't going to be fixing them anyway.

In any case, the fix is really trivial, harmless and seems to be the correct thing to do anyway: replace `pytest.raises` with `pytest.warns`, since it's warnings the test is checking for after all.

I'll submit a PR shortly.

Environment:

- Python version: 3.10.0b3
- Werkzeug version: eff04478a83619b4d7f15e6eee16a99bd80ed879

--- a/tests/test_wrappers.py
+++ b/tests/test_wrappers.py
@@ -1613,7 +1613,7 @@ def test_request_mixins_deprecated(cls):
     class CheckRequest(cls, wrappers.Request):
         pass
 
-    with pytest.warns(DeprecationWarning, match=cls.__name__):
+    with pytest.deprecated_call(match=cls.__name__):
         CheckRequest({"SERVER_NAME": "example.org", "SERVER_PORT": "80"})
 
 
@@ -1633,29 +1633,29 @@ def test_response_mixins_deprecated(cls):
     class CheckResponse(cls, wrappers.Response):
         pass
 
-    with pytest.raises(DeprecationWarning, match=cls.__name__):
+    with pytest.deprecated_call(match=cls.__name__):
         CheckResponse()
 
 
 def test_check_base_deprecated():
-    with pytest.raises(DeprecationWarning, match=r"issubclass\(cls, Request\)"):
+    with pytest.deprecated_call(match=r"issubclass\(cls, Request\)"):
         assert issubclass(wrappers.Request, wrappers.BaseRequest)
 
-    with pytest.raises(DeprecationWarning, match=r"isinstance\(obj, Request\)"):
+    with pytest.deprecated_call(match=r"isinstance\(obj, Request\)"):
         assert isinstance(
             wrappers.Request({"SERVER_NAME": "example.org", "SERVER_PORT": "80"}),
             wrappers.BaseRequest,
         )
 
-    with pytest.raises(DeprecationWarning, match=r"issubclass\(cls, Response\)"):
+    with pytest.deprecated_call(match=r"issubclass\(cls, Response\)"):
         assert issubclass(wrappers.Response, wrappers.BaseResponse)
 
-    with pytest.raises(DeprecationWarning, match=r"isinstance\(obj, Response\)"):
+    with pytest.deprecated_call(match=r"isinstance\(obj, Response\)"):
         assert isinstance(wrappers.Response(), wrappers.BaseResponse)
 
 
 def test_response_freeze_no_etag_deprecated():
-    with pytest.raises(DeprecationWarning, match="no_etag"):
+    with pytest.deprecated_call(match="no_etag"):
         Response("Hello, World!").freeze(no_etag=True)
 
 

